src/Form/FormAutosuggest.jsx

```diff
@@ -14,292 +15,358 @@ import Spinner from '../Spinner';
 
-    setIsMenuClosed(true);
+    const handleMenuItemFocus = (menuItemId) => {
+      setActiveMenuItemId(menuItemId);
+    };
 
-    if (onClick) {
-      onClick(e);
-    }
-  };
+    const collapseDropdown = () => {
+      setDropdownItems([]);
+      setDropdownExpanded(false);
+      setActiveMenuItemId(null);
+    };
 
-  function getItems(strToFind = '') {
-    let childrenOpt = React.Children.map(children, (child) => {
-      // eslint-disable-next-line no-shadow
-      const { children, onClick, ...rest } = child.props;
-      const menuItemId = uuidv4();
-
-      return React.cloneElement(child, {
-        ...rest,
-        children,
-        'data-value': children,
-        onClick: (e) => handleItemClick(e, onClick),
-        id: menuItemId,
-        onFocus: () => handleMenuItemFocus(menuItemId),
-      });
-    });
+    const handleItemSelect = (e, onClick) => {
+      const selectedValue = e.currentTarget.getAttribute('data-value');
+      const selectedId = e.currentTarget.id;
 
-    if (strToFind.length > 0) {
-      childrenOpt = childrenOpt
-        .filter((opt) => (opt.props.children.toLowerCase().includes(strToFind.toLowerCase())));
-    }
+      setHasValue(true);
+      setHasSelection(true);
+      setDisplayValue(selectedValue);
 
-    return childrenOpt;
-  }
+      if (onChange && (!value || (value && selectedValue !== value.selectionValue))) {
+        onChange({
+          userProvidedText: selectedValue,
+          selectionValue: selectedValue,
+          selectionId: selectedId,
+        });
+      }
 
-  const handleExpand = () => {
-    setIsMenuClosed(!isMenuClosed);
+      collapseDropdown();
 
-    const newState = {
-      dropDownItems: [],
+      if (onClick) {
+        onClick(e);
+      }
     };
 
-    if (isMenuClosed) {
-      setIsActive(true);
-      newState.dropDownItems = getItems(state.displayValue);
-      newState.errorMessage = '';
-    }
-
-    setState(prevState => ({
-      ...prevState,
-      ...newState,
-    }));
-  };
-
-  const iconToggle = (
-    <IconButton
-      className="pgn__form-autosuggest__icon-button"
-      data-testid="autosuggest-iconbutton"
-      tabindex="-1"
-      src={isMenuClosed ? KeyboardArrowDown : KeyboardArrowUp}
-      iconAs={Icon}
-      size="sm"
-      variant="secondary"
-      alt={isMenuClosed
-        ? intl.formatMessage(messages.iconButtonOpened)
-        : intl.formatMessage(messages.iconButtonClosed)}
-      onClick={(e) => handleExpand(e, isMenuClosed)}
-    />
-  );
-
-  const leaveControl = () => {
-    setIsActive(false);
-
-    setState(prevState => ({
-      ...prevState,
-      dropDownItems: [],
-      errorMessage: !state.displayValue ? errorMessageText : '',
-    }));
+    function getItems(strToFind = '') {
+      let childrenOpt = React.Children.map(children, (child) => {
+      // eslint-disable-next-line no-shadow
+        const { children, onClick, ...rest } = child.props;
```

Comment:
nit: to avoid `// eslint-disable-next-line no-shadow`, consider renaming `children` once deconstructed (i.e., `const { children: childChildren, ... } = child.props;` to avoid the naming conflict with the parent `children`.

Reply:
addressed https://github.com/openedx/paragon/compare/f8a4e5ee118199936c3ff6653e3dfb389d12fffa..9310aa04d988eba1ee524cf920e5ad54dd39590f#diff-96e13c1a773ec8d218ea5667d3dec6c8f6eeadcb00c81584f50d6c3d4d52135fR90